src/set/mod.rs

```diff
@@ -0,0 +1,41 @@
+//! A concurrent hash set backed by FlurryHashMap.
+
+use crossbeam::epoch;
+use std::collections::hash_map::RandomState;
+use std::hash::Hash;
+
+use crate::FlurryHashMap;
+
+/// A concurrent hash set.
+#[derive(Debug)]
+pub struct FlurryHashSet<T, S = RandomState> {
+    map: FlurryHashMap<T, (), S>,
+}
+
+impl<T> FlurryHashSet<T, RandomState>
+where
+    T: Sync + Send + Clone + Hash + Eq,
+{
+    /// Creates a new, empty set with the default initial table size (16).
+    pub fn new() -> Self {
+        Self {
+            map: FlurryHashMap::<T, ()>::new(),
+        }
+    }
+
+    /// Adds a value to the set.
+    ///
+    /// If the set did not have this value present, true is returned.
+    ///
+    /// If the set did have this value present, false is returned.
+    pub fn insert(&self, value: T) -> bool {
```

Comment:
I'm a little unclear as to the semantics of when a guard needs to be passed in vs constructed inside the public API. AFAICT it's to do with the lifetime of the returned value, and since we're not returning anything here I think this is okay?

Reply:
That is exactly correct -- the guard is needed to ensure that the value a reference we return points to lives for as long as the reference does. Since we here do not return any reference, we do not have to take a guard in either, since it's fine for the lifetime of `old` to not outlast the current function (which is where `guard` will be dropped).